qubesusbproxy/tests.py

```diff
@@ -722,12 +722,13 @@ def test_010_on_qdb_change_multiple_assignments_including_full(self):
             qubesusbproxy.core3ext.USBDevice(Port(back, "1-1", "usb"))
         )
 
-        self.ext.attach_and_notify = AsyncMock()
         loop = asyncio.get_event_loop()
-        loop.run_until_complete(self.ext.on_domain_start(front, None))
-        self.assertEqual(
-            self.ext.attach_and_notify.call_args[0][1].options, {"pid": "did"}
-        )
+        with (mock.patch.object(self.ext, "attach_and_notify")
+              as attach_and_notify):
```

Comment:
I think you need to mock it with `AsyncMock()` (just add as the third arg to patch.object), otherwise you'll get back to the TypeError.

Reply:
unittest mock automatically uses `AsyncMock` for coroutines

Reply:
Ah, nice :)